src/StreamJsonRpc/JsonRpcTargetOptions.cs

```diff
@@ -0,0 +1,34 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace StreamJsonRpc
+{
+    using System;
+
+    /// <summary>
+    /// Options that may customize how a target object is added to a <see cref="JsonRpc"/> instance.
+    /// </summary>
+    public class JsonRpcTargetOptions
+    {
+        /// <summary>
+        /// Gets or sets a function that takes the CLR method name and returns the RPC method name.
+        /// This method is useful for adding prefixes to all methods, or making them camelCased.
+        /// </summary>
+        public Func<string, string> MethodNameTransform { get; set; }
+
+        /// <summary>
+        /// Gets or sets a value indicating whether events raised on the target object
+        /// should be relayed to the client via a JSON-RPC notify message.
+        /// </summary>
+        /// <value>The default is <c>true</c>.</value>
+        public bool NotifyClientOfEvents { get; set; } = true;
+
+        /// <summary>
+        /// Gets an instance with default properties.
+        /// </summary>
+        /// <remarks>
+        /// Callers should *not* mutate properties on this instance since it is shared.
```

Comment:
You could introduce an interface that only had the getters for MethodNameTransform and NotifyClientOfEvents that this property returns to guarantee that it isn't mutated.

Reply:
I would do that, or introduce a "frozen" field that property setters would throw if true for. But since this is an internal member, and currently only used in one place, solving the problem seemed overkill.